LanguageFeatures/Private-fields-promotion/promotion_A05_t04.dart

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion An instance field is promotable only if
+/// (a) it is private,
+/// (b) it is final, and
+/// (c) all other concrete instance getters with the same name in the same
+/// library are also final fields
+///
+/// @description Checks that if there are other concrete instance getters with
+/// the same name in the same library and they are not a final fields then the
+/// field is not promotable. Test the case when there is a getter with the same
+/// name in some extension in the same library
```

Comment:
Hmm, I had not thought about this case.  I'm ok with leaving the test as is for now, but I've filed https://github.com/dart-lang/language/issues/2404 to check if anyone on the language team feels strongly that this should behave differently.